### PR TITLE
3.0 Allow for custom finder name in Table::get().

### PR DESCRIPTION
**Pull request by robertpustulka**

### CODE REVIEW

File: src/ORM/Table.php [hunk 1]
```diff
@@ -1158,9 +1158,10 @@ public function get($primaryKey, $options = [])
 
         $cacheConfig = isset($options['cache']) ? $options['cache'] : false;
         $cacheKey = isset($options['key']) ? $options['key'] : false;
-        unset($options['key'], $options['cache']);
+        $finderName = isset($options['finderName']) ? $options['finderName'] : 'all';
```
Comment (ADmad) [inline src/ORM/Table.php]: Better to use key name `finder` for consistency with PaginatorComponent.

**Comment by lorenzo** [comment]
I don't think this is a good idea, what is your use case?

**Comment by robertpustulka** [comment]
I'm trying to get an entity with translations in my plugin but only as an
option.

W dniu wtorek, 21 lipca 2015 José Lorenzo Rodríguez <
notifications@github.com> napisał(a):

> I don't think this is a good idea, what is your use case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cakephp/cakephp/pull/7079#issuecomment-123350761>.
>


-- 
Pozdrawiam,
Robert Pustułka

**Comment by robertpustulka** [comment]
Is it ok to be merged? It would help me develop my plugin...

**Comment by lorenzo** [comment]
I'm not really convinced this is a good idea. Can you explain how this feature helps developing your plugin?

**Comment by robertpustulka** [comment]
@lorenzo I'm developing a plugin providing `ClonableBehavior`, which allows to create exact copies of entities. It calls `get()` on a table, copies retrieved entity, clears its PK and returns cloned entity.
This plugin allows also to clone entities of translated tables. Behavior detects if a table has a `TranslateBehavior` and whether it needs to retrieve all translations for an entity. Its easier for me to just pass optional finder name to `get()` method instead of rewriting it only to get an entity using `translate` finder.

**Comment by lorenzo** [comment]
What about finders that require options be passed?

**Comment by robertpustulka** [comment]
@lorenzo `get()` passes its `$options` into `find()`.

**Comment by lorenzo** [comment]
Fair enough, I think it makes sense. Would you mind documenting the new feature in the book? @robertpustulka

**Comment by robertpustulka** [comment]
Can this be merged into 3.1?

**Pull request merged**